File: 74751_lab4_task3.py
import math

def calculate_stats(hemoglobin_levels):
    if not hemoglobin_levels:
        return 0, 0
    n = len(hemoglobin_levels)
    mean = sum(hemoglobin_levels) / n
    variance = sum((x - mean) ** 2 for x in hemoglobin_levels) / n
    std_dev = math.sqrt(variance)
    return mean, std_dev

def insert_test_result(results, new_result, index):
    if 0 <= index <= len(results):
        results.insert(index, new_result)
        return True
    return False

def assess_risk(patient_id, cholesterol, blood_pressure, fasting_sugar):
    risk = 0
    if cholesterol > 200: risk += 1
    if blood_pressure > 130: risk += 1
    if fasting_sugar > 100: risk += 1
    
    if risk >= 2:
        return f"Patient {patient_id}: High Risk"
    elif risk == 1:
        return f"Patient {patient_id}: Moderate Risk"
    return f"Patient {patient_id}: Low Risk"

if __name__ == "__main__":
    hemo = [13.5, 14.2, 12.8, 15.0]
    avg, std = calculate_stats(hemo)
    print(f"Average: {avg:.2f}, Std Dev: {std:.2f}")
    
    success = insert_test_result(hemo, 14.5, 2)
    print(f"Insert Success: {success}, New list: {hemo}")
    
    risk = assess_risk("P123", 220, 140, 95)
    print(risk)

File: 74751_lab4_task3.cpp
#include <iostream>
#include <vector>
#include <cmath>
#include <string>

using namespace std;

void calculate_stats(const vector<double>* hemoglobin_levels, double& average, double& std_dev) {
    if (hemoglobin_levels->empty()) {
        average = 0;
        std_dev = 0;
        return;
    }
    double sum = 0;
    for (double val : *hemoglobin_levels) {
        sum += val;
    }
    average = sum / hemoglobin_levels->size();
    
    double var_sum = 0;
    for (double val : *hemoglobin_levels) {
        var_sum += (val - average) * (val - average);
    }
    std_dev = sqrt(var_sum / hemoglobin_levels->size());
}

bool insert_test_result(vector<double>* results, double new_result, int index) {
    if (index >= 0 && index <= results->size()) {
        results->insert(results->begin() + index, new_result);
        return true;
    }
    return false;
}

string assess_risk(string patient_id, double cholesterol, double blood_pressure, double fasting_sugar) {
    int risk = 0;
    if (cholesterol > 200) risk++;
    if (blood_pressure > 130) risk++;
    if (fasting_sugar > 100) risk++;
    
    if (risk >= 2) return "Patient " + patient_id + ": High Risk";
    if (risk == 1) return "Patient " + patient_id + ": Moderate Risk";
    return "Patient " + patient_id + ": Low Risk";
}

int main() {
    vector<double> hemo = {13.5, 14.2, 12.8, 15.0};
    double avg, std;
    calculate_stats(&hemo, avg, std);
    cout << "Average: " << avg << ", Std Dev: " << std << endl;
    
    bool success = insert_test_result(&hemo, 14.5, 2);
    cout << "Insert Success: " << (success ? "true" : "false") << ", New list: ";
    for (double v : hemo) cout << v << " ";
    cout << endl;
    
    cout << assess_risk("P123", 220, 140, 95) << endl;
    
    return 0;
}
